### ISSUE TITLE
Small values are being cut off to 0 during optimizer pass in training

### ISSUE THREAD

**Issue by umalesTT** [tenstorrent/tt-xla#342]
During model training, in optimizer pass, when tensor of calculated gradients is multiplied by learning rate, part of the values in the resulting tensor are cut to 0, resulting in some weights not being updated at all - which then affects model accuracy etc.

The reason for this is https://github.com/tenstorrent/tt-metal/issues/18955.
In tests, we might want to cover cases similar to this, i.e. operations with smaller values (than the ones we have now) because otherwise issues like this might go under the radar and because in training we usually operate with small values.

**Comment by umalesTT** [comment]
Closed with https://github.com/tenstorrent/tt-metal/pull/26101
